### PR TITLE
gadgets/trace_open/test: Verify container name and ID

### PR DESCRIPTION
**Pull request by blanquicet**
# Verify container name and ID in trace_open gadget test

- Leverage the [containers.TestContainer](https://github.com/inspektor-gadget/inspektor-gadget/blob/97eaf2fc80da992a6ed1b54cebc861bbf91844bd/pkg/testing/containers/factory.go#L45-L48) methods to retrieve the container ID and verify it in the expected output.
- Specify the image we want to use and verify it in the expected output.
- NOTE: The image digest is not yet enriched for Docker. See https://github.com/inspektor-gadget/inspektor-gadget/issues/2365.

## Testing done

CI is green 🙂 

## TODO (Future PR)

The [containers.TestContainer](https://github.com/inspektor-gadget/inspektor-gadget/blob/97eaf2fc80da992a6ed1b54cebc861bbf91844bd/pkg/testing/containers/factory.go#L45-L48) also provides the `Pid()`. Using that value, we can retrieve the mount ns inode as we do in [getNsInode](https://github.com/inspektor-gadget/inspektor-gadget/blob/97eaf2fc80da992a6ed1b54cebc861bbf91844bd/internal/test/runner.go#L205-L218) and verify also the `MountNsID` field in the test.

### CODE REVIEW

File: gadgets/trace_open/test/trace_open_test.go [hunk 1]
```diff
@@ -46,14 +46,18 @@ type traceOpenEvent struct {
 func TestTraceOpen(t *testing.T) {
 	gadgettesting.RequireEnvironmentVariables(t)
 
-	cn := "test-trace-open"
 	containerFactory, err := containers.NewContainerFactory("docker")
 	require.NoError(t, err, "new container factory")
 
+	cn := "test-trace-open"
```
Comment (mauriciovasquezbernal) [inline gadgets/trace_open/test/trace_open_test.go:54]: what about using containerName? I always think of Cartoon Network when reading cn...

File: gadgets/trace_open/test/trace_open_test.go [hunk 2]
```diff
@@ -46,20 +47,36 @@ type traceOpenEvent struct {
 func TestTraceOpen(t *testing.T) {
 	gadgettesting.RequireEnvironmentVariables(t)
 
-	cn := "test-trace-open"
-	containerFactory, err := containers.NewContainerFactory("docker")
+	runtime := "docker"
```
Comment (mauriciovasquezbernal) [inline gadgets/trace_open/test/trace_open_test.go:50]: I suppose it's fairly simple to allow this to be configurable (in another PR for sure). Using an env variable seems the easiest way, right?

**Review by mauriciovasquezbernal — APPROVED** [review]
Looks great from code inspection. I really like where the testing is going.

File: gadgets/trace_open/test/trace_open_test.go (hunk 2 above)
Comment (blanquicet) [reply gadgets/trace_open/test/trace_open_test.go:50]: Yep, that's why I made that change. To prepare everything to make it easily configurable.

Comment (blanquicet) [reply gadgets/trace_open/test/trace_open_test.go:50]: I'm still not sure if I prefer using env variable to pass this kind of configuration of flags, like [here](https://github.com/inspektor-gadget/inspektor-gadget/blob/d54838472c7a2be0894f90369bb3068e630415d9/integration/ig/non-k8s/integration_test.go#L32).

Comment (mauriciovasquezbernal) [reply gadgets/trace_open/test/trace_open_test.go:50]: I think the flag doesn't work well as we'd need to have a `TestMain` function to parse the flag, then running the test alone (in the gadget folder) won't work.

**Pull request merged**